### ISSUE TITLE
Use repo-infra in kubernetes/kubernetes

### ISSUE THREAD

**Issue by nikhita** [kubernetes/repo-infra#97]
@hoegaarden and @mariantalla have been working on improving the scripts in https://github.com/kubernetes/repo-infra/pull/94 and https://github.com/kubernetes/repo-infra/pull/82. (thank you so much for those btw! :smile: )

To make sure we make full use of these improvements, we'll need to use k/repo-infra in k/k. This issue is meant as a place to discuss if we want to do so (if not, any improvements in scripts should be redirected to k/k) and if yes, what challenges we currently face to make it possible.

/cc @ixdy @mikedanese @cblecker 

/sig contributor-experience
repo-infra is contribex subproject

**Comment by nikhita** [comment]
I wish we had a sig-hack-bash-fire...

**Comment by hoegaarden** [comment]
TL;DR: Can we just merge #82 and #94?

As far as I know nothing has happened here or somewhere else regarding #94 & #82, right?

My feeling is that
- no one will care about this until we open a PR to `k/k` (or a similar prominent repo) and spark a discussion there
  - alternative: Write a KEP for shared verify / update tools
- we would not make anything worse when merging those PRs here, `repo-infra` will stay very much the same as it is right now for any of the already existing users
  - if that is not the case, please let me know!

I think there is really just [one major open question](https://github.com/kubernetes/repo-infra/pull/94#issuecomment-455349615) (paraphrased):
> Do we want to introduce another dependency management mechanism, `git subtree`, and advocate that?

... which I'd answer with:
- at least it (how to get the thing in with `git subtree`) is documented
- anything is better then what is happening now (every repo  slighty different verify scripts, eg. for the boilerplate stuff, every new repo probably just `cp`s stuff from the next kubernetes repo in reach)
- we have verify / update mechanisms (that could be enforced by CI)
- if people prefer `git submodule` or any other mechanism to pull in repo-infra, they still can do it, we don't break that

What do you all thing about my random ramblings?

**Comment by BenTheElder** [comment]
this issue title is slightly confusing since we _do_ use some of repo-infra in kubernetes 😅
